### PR TITLE
Create node.js.yml

### PR DESCRIPTION
**Pull request by kswang2400**
create javascript ci tool that lints, builds, and tests the ibr package before landing to master

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      matrix:
+        node-version: [10.x, 12.x, 14.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
```
Comment (charbull) [inline .github/workflows/node.js.yml]: you can run it only your directory check the `working_directory`

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
```
Comment (charbull) [inline .github/workflows/node.js.yml:7]: same here and on pull request you can target only your folder

Comment (kswang2400) [reply .github/workflows/node.js.yml:7]: nice, ty ty

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (kswang2400) [reply .github/workflows/node.js.yml]: nice ty ty

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+
+    - name: Run Google es-linter
+      run: npm run lint
```
Comment (charbull) [inline .github/workflows/node.js.yml]: you can use the google js styleguide from here: https://github.com/google/styleguide

I got the pylint from here: https://github.com/google/styleguide/blob/ea180f7259da4ae90ae086c1eefbeb5f440dbea2/pylintrc

Comment (kswang2400) [reply .github/workflows/node.js.yml:29]: yeah, trying to use "eslint-config-google": "^0.14.0", from ibr package but there's some errors

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
+    branches: [ master ]
```
Comment (charbull) [inline .github/workflows/node.js.yml:8]: awesome ! you can specify here the working directory, when the push or pr matches the path then the action is triggered, do you want to add the part to match the file path?
I need to add it as well on the other actions

**Pull request merged**